src/Psl/Class/is_readonly.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Class;
+
+use ReflectionClass;
+
+/**
+ * Checks if class is read only.
+ *
+ * @param class-string $class_name
+ */
+function is_readonly(string $class_name): bool
+{
+    /** @psalm-suppress MissingThrowsDocblock */
+    return (new ReflectionClass($class_name))->isReadOnly();
```

Comment:
oh, did not know about the internals of the Class component : looks like you were already taking the functional approach there ;)

Reply:
`Class`, `Trait`, and `Interface` are considered much simpler than the reflection API. they only provide a handful of fucntions.